Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -259,6 +260,15 @@ class BrowserViewController: UIViewController {
             toolbar?.applyUIMode(isPrivate: tabManager.selectedTab?.isPrivate ?? false)
             toolbar?.applyTheme()
             toolbar?.addNewTabButton.isHidden = true
+            // This is for showing (+) add tab middle button with A/B test where we need to update both toolbar and url bar when (+) button is enabled.
+            // Urlbar already has reader mode state but we still need to refresh it so that if reader mode is available we don't accidently show reload or stop button in url bar
+            if shouldShowNewTabButton {
+                toolbar?.updateMiddleButtonState(.newTab)
+                let state = urlBar.locationView.readerModeState
+                urlBar.updateReaderModeState(state)
+            } else {
+                toolbar?.updateMiddleButtonState(currentMiddleButtonState ?? .search)
```

Comment:
Added this part to make sure when we come from background to foreground that we have proper middle button state. This is also helpful when user rotates their device and will fix - https://github.com/mozilla-mobile/firefox-ios/issues/7167